backend/sublet/serializers.py

```diff
@@ -74,29 +99,44 @@ def parse_amenities(self, raw_amenities):
 
     def create(self, validated_data):
         validated_data["subletter"] = self.context["request"].user
+        images = validated_data.pop("images")
         instance = super().create(validated_data)
         data = self.context["request"].POST
         amenities = self.parse_amenities(data.getlist("amenities"))
         instance.amenities.set(amenities)
         instance.save()
+        # TODO: make this atomic
+        for img in images:
+            img_serializer = SubletImageSerializer(data={"sublet": instance.id, "image": img})
+            img_serializer.is_valid(raise_exception=True)
+            img_serializer.save()
         return instance
 
+    # delete_images is a list of image ids to delete
     def update(self, instance, validated_data):
         # Check if the user is the subletter before allowing the update
-        if (
-            self.context["request"].user == instance.subletter
-            or self.context["request"].user.is_superuser
-        ):
-            amenities_data = self.context["request"].data
-            if amenities_data.get("amenities") is not None:
-                amenities = self.parse_amenities(amenities_data.getlist("amenities"))
-                instance.amenities.set(amenities)
-            validated_data.pop("amenities", None)
-            instance = super().update(instance, validated_data)
-            instance.save()
-        else:
-            raise serializers.ValidationError("You do not have permission to update this sublet.")
-
+        # This is probably redundant given permissions classes?
+        # if (
+        #     self.context["request"].user == instance.subletter
+        #     or self.context["request"].user.is_superuser
+        # ):
+        amenities_data = self.context["request"].data
+        if amenities_data.get("amenities") is not None:
+            amenities = self.parse_amenities(amenities_data["amenities"])
+            instance.amenities.set(amenities)
+        validated_data.pop("amenities", None)
+        delete_images = validated_data.pop("delete_images")
+        instance = super().update(instance, validated_data)
+        instance.save()
+        existing_images = Sublet.objects.get(id=instance.id).images.all()
+        print(existing_images)
+        for img in delete_images:
+            get_object_or_404(existing_images, id=img)
```

Comment:
same thing here, do all validation before we start saving to the database, also would recommend a map or list comprehension here
```
delete_images = map(delete_images, lambda id: get_object_or_404(existing_images, id=id)
# OR
delete_images = [get_object_or_404(existing_images, id=img) for img in delete_images]
```
also probably accept the parameter name as `delete_images_ids` since that's more accurate